### PR TITLE
Use URL parameters to visit specified trace

### PR DESCRIPTION
**Pull request by QHWG67**
People want to use the trace URL parameters to visit the specified trace information instead of using Search button. It is very useful for some tools to use hyperlink to jump to skywalking trace page via endpointname and trace ID. This update is for URL parameters to visite specified trace.
example:
http://localhost:8880/trace?endpointname=runtest&traceid=2.30.15753618423830000

### CODE REVIEW

**Comment by wu-sheng** [comment]
Good feature, I will ask related committers to review. I like this.

File: src/views/components/trace/trace-search.vue [hunk 1]
```diff
@@ -95,12 +95,19 @@
...
     private instance: Option = {label: 'All', key: ''};
-    private endpointName: string = localStorage.getItem('endpointName') || '';
-    private traceId: string = localStorage.getItem('traceId') || '';
+    private endpointName: string = this.getUrlKey('endpointname') || localStorage.getItem('endpointName') || '';
+    private traceId: string = this.getUrlKey('traceid') || localStorage.getItem('traceId') || '';
     private traceState: Option = {label: 'All', key: 'ALL'};
     @Prop({default: false, type: Boolean})
     private inTopo!: boolean;
 
+    private getUrlKey(name : any) {
```
Comment (x22x22) [inline src/views/components/trace/trace-search.vue]: It doesn't need to be that complicated, please use "this.$route.query". ref: https://router.vuejs.org/guide/essentials/dynamic-matching.html#dynamic-route-matching

Comment (QHWG67) [reply src/views/components/trace/trace-search.vue]: @x22x22 Thanks for your review! Based on your comments, I have updated the codes, pls continue to review and merge.

**Comment by QHWG67** [comment]
@x22x22 and all, the new version is updated for your review again, thanks.

File: src/views/components/trace/trace-search.vue [hunk 2]
```diff
@@ -236,6 +236,10 @@
     }
 
     private created() {
+      const endpoint: any = this.$route.query.endpointname || this.endpointName;
```
Comment (x22x22) [inline src/views/components/trace/trace-search.vue]: Would it be better to write this way?

```js
this.endpointName = this.$route.query.endpointname ? this.$route.query.endpointname : this.endpointName
this.traceId = this.$route.query.traceid? this.$route.query.traceid: this.traceId
```

Comment (wu-sheng) [reply src/views/components/trace/trace-search.vue]: @QHWG67 I think the above one is more clear than `||`.

Comment (QHWG67) [reply src/views/components/trace/trace-search.vue]: @x22x22 @wu-sheng per your requested, they are updated. Pls check.

Notes:
by default, this.$route.query.endpointname is the string array (it means the URL parameter may have duplicated key name, this should not be happen normally), so it cannot directly be "this.endpointName = this.$route.query.endpointname", then get error as below, so here has to use toString() and get all values of one key name, user may find their url problem and adjust url accordingly base on result. This update is verified by me before commiting.
252:5 Type 'string | string[]' is not assignable to type 'string'.
  Type 'string[]' is not assignable to type 'string'.
    250 |
    251 |     private created() {
  > 252 |       this.endpointName = this.$route.query.endpointname? this.$route.query.endpointname : this.endpointName;

Comment (wu-sheng) [reply src/views/components/trace/trace-search.vue]: Make sense. In theory, people could set duplicated keys in the URL parameters. It is a list, not a map. I think `toString` is fine. 

I am going to merge this. Could you submit the submodule sync to main repo after I am done?

**Comment by wu-sheng** [comment]
@QHWG67 merged, could you submit a submodule sync PR to the main repo?

**Comment by QHWG67** [comment]
> @QHWG67 merged, could you submit a submodule sync PR to the main repo?

@wu-sheng ok, thx! I will do.

**Comment by QHWG67** [comment]
> > @QHWG67 merged, could you submit a submodule sync PR to the main repo?
> 
> @wu-sheng ok, thx! I will do.

@wu-sheng Request is submitted : https://github.com/apache/skywalking/pull/4037

**Pull request merged**